cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -101,3 +107,37 @@ func makeModelKeyValue(agentTags map[string]string) []model.KeyValue {
 
 	return tags
 }
+
+func makeSliceOfTags(includedTags string) []string {
+	if includedTags != "" {
+		tags := strings.Split(strings.Trim(includedTags, " "), ",")
+		return tags
+	} else {
+		return make([]string, 0)
+	}
+}
+
+//remove the tags not present in filterProcessTags flag
+func pruneUnwantedTags(filterProcessTagsSlice []string, spans []*model.Span, process *model.Process) []*model.Span {
+	for _, span := range spans {
+		if span.Process != nil && span.Process.Tags != nil {
+			finalTags := make([]model.KeyValue, 0)
+			for _, tag := range span.Process.Tags {
+				if contains(filterProcessTagsSlice, tag.Key) {
```

Comment:
the sanitizer should convert (during initialization) the array of tag names into a `map[string]struct{}` for faster lookup